### ISSUE TITLE
Detect and error out on infinite loop

### ISSUE THREAD

**Issue by relaxnow** [OpenConext/OpenConext-engineblock#206]
Sometimes SPs will infinitely redirect a user to EB, EB should detect this and show the user that something is wrong with the SP to try again or contact the SP.

Related: https://svn.surfnet.nl/svn/aai/sfs/phpFederate/trunk/lib/session.php line 170

**Comment by thijskh** [comment]
This would be a nice enhancement indeed.

**Comment by thijskh** [comment]
The svn server that @relaxnow references has since been shut down, but the linked code fragment is:

```php
         } else {
                    $_SESSION['dos-protect'][] = time();
                    $now = time();
                    while (($_SESSION['dos-protect'][0]) < ($now - 15)) array_shift($_SESSION['dos-protect']);
                    if (count($_SESSION['dos-protect']) > 5) {
                            throw new Exception('SSO loop detected: probably the service you are accessing is misconfigured!');
                    }
                    /* SSO succeeded: return to the caller SP */
                    session_return('updated');
         }
```

**Comment by DRvanR** [comment]
Resolved through #365
